components/invitation_modal/invite_view.tsx

```diff
@@ -0,0 +1,339 @@
+import {InviteToTeamTreatments} from 'mattermost-redux/constants/config';
+import {Channel} from 'mattermost-redux/types/channels';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {Team} from 'mattermost-redux/types/teams';
+
+import {getSiteURL} from 'utils/url';
+import {Constants} from 'utils/constants';
+
+import {trackEvent} from 'actions/telemetry_actions';
+import {getAnalyticsCategory} from 'components/next_steps_view/step_helpers';
+import useCopyText from 'components/common/hooks/useCopyText';
+import UsersEmailsInput from 'components/widgets/inputs/users_emails_input';
+import UpgradeLink from 'components/widgets/links/upgrade_link';
+import NotifyLink from 'components/widgets/links/notify_link';
+import {t} from 'utils/i18n.jsx';
+import {SubscriptionStats} from 'mattermost-redux/types/cloud';
+
+import AddToChannels, {CustomMessageProps, InviteChannels, defaultCustomMessage, defaultInviteChannels} from './add_to_channels';
+import InviteAs, {As} from './invite_as';
+import './invite_view.scss';
+
+export const defaultInviteState: InviteState = deepFreeze({
+    as: As.MEMBER,
+    customMessage: defaultCustomMessage,
+    sending: false,
+    inviteChannels: defaultInviteChannels,
+    usersEmails: [],
+    usersEmailsSearch: '',
+});
+
+export type InviteState = {
+    customMessage: CustomMessageProps;
+    as: As;
+    sending: boolean;
+    inviteChannels: InviteChannels;
+    usersEmails: Array<UserProfile | string>;
+    usersEmailsSearch: string;
+};
+
+export type Props = InviteState & {
+    setInviteAs: (as: As) => void;
+    invite: () => void;
+    onChannelsChange: (channels: Channel[]) => void;
+    onChannelsInputChange: (channelsInputValue: string) => void;
+    onClose: () => void;
+    currentTeam: Team;
+    currentChannelName: string;
+    inviteToTeamTreatment: InviteToTeamTreatments;
+    setCustomMessage: (message: string) => void;
+    toggleCustomMessage: () => void;
+    channelsLoader: (value: string, callback?: (channels: Channel[]) => void) => Promise<Channel[]>;
+    regenerateTeamInviteId: (teamId: string) => void;
+    isAdmin: boolean;
+    usersLoader: (value: string, callback: (users: UserProfile[]) => void) => Promise<UserProfile[]> | undefined;
+    onChangeUsersEmails: (usersEmails: Array<UserProfile | string>) => void;
+    isCloud: boolean;
+    subscriptionStats?: SubscriptionStats | null;
+    cloudUserLimit: string;
+    emailInvitationsEnabled: boolean;
+    onUsersInputChange: (usersEmailsSearch: string) => void;
+    headerClass: string;
+    footerClass: string;
+    canInviteGuests: boolean;
+    canAddUsers: boolean;
+}
+
+function clearMinHeight() {
+    const modalBody = document.querySelector('.modal-body') as HTMLDivElement;
+    if (!modalBody) {
+        return;
+    }
+    if (modalBody.style.minHeight) {
+        modalBody.style.minHeight = '';
+    }
+}
+
+const FOOTER_HEIGHT = 63;
+
+const ensureInvitesVisible = debounce((options: UserProfile[]) => {
```

Comment:
`ensureInvitesVisible` and `clearMinHeight` was the best way I could think of to overcome the `overflow: hidden` hidden restriction on modal body content not being able to display outside of modal body. In the fullscreen modal, this was not a problem because the modal had the whole screen to work with. It makes some effort to not make the modal taller than the screen.

clearMinHeight *should* be invoked everywhere so we don't get a goofily long looking modal once the dropdown closes or has no options, but extra scrutiny is warranted here.